hipchat/hipchat.go

```diff
@@ -275,6 +325,58 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	return resp, err
 }
 
+func (c *Client) doWithRetryPolicy(req *http.Request, policy RetryPolicy) (*http.Response, error) {
+	currentTry := 0
+
+	for willContinue(currentTry, policy) {
+		currentTry = currentTry + 1
+		resp, err := c.client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		c.captureRateLimits(resp)
+		if code := resp.StatusCode; code == http.StatusTooManyRequests {
+			if willContinue(currentTry, policy) {
+				doDelay(currentTry, policy)
+			}
+		} else {
+			return resp, nil
+		}
+	}
+	return nil, fmt.Errorf("max retries exceeded (%d)", policy.MaxRetries)
+}
+
+func willContinue(currentTry int, policy RetryPolicy) bool {
+	return currentTry <= policy.MaxRetries
+}
+
+func doDelay(currentTry int, policy RetryPolicy) {
+	jitter := time.Duration(rand.Int63n(2*int64(policy.JitterDelay))) - policy.JitterBias
+	linearDelay := time.Duration(currentTry)*policy.MinDelay + jitter
+	if linearDelay > policy.MaxDelay {
+		linearDelay = policy.MaxDelay
+	}
+	time.Sleep(time.Duration(linearDelay))
+}
+
+func setIfPresent(src string, dest *int) {
+	if len(src) > 0 {
+		v, err := strconv.Atoi(src)
+		if err != nil {
+			*dest = v
+		}
+	}
```

Comment:
When the header is not present, `Client` will keep the RateLimit/FloodControl values of the previous response with that headers. Is this expected ?